### PR TITLE
[ADDED] JetStream: Nak delay and BackOff lists

### PR DESCRIPTION
**Pull request by kozlovic**
Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: test/js_test.go [hunk 1]
```diff
@@ -2741,6 +2741,100 @@ func TestJetStreamSubscribe_AckPolicy(t *testing.T) {
...
+		js.Publish("bar", []byte("msg"))
+		sub, err := js.SubscribeSync("bar", nats.Durable("nak_dur"))
+		if err != nil {
+			t.Fatalf("Error on subscribe: %v", err)
+		}
+		msg, err := sub.NextMsg(time.Second)
+		if err != nil {
+			t.Fatalf("Error on NextMsg: %v", err)
+		}
+		if err := msg.Nak(nats.NakDelay(500 * time.Millisecond)); err != nil {
```
Comment (derekcollison) [inline test/js_test.go:2755]: Hmm.. Looking at this suggest maybe NakWithDelay(d time.Duration) might be better? WDYT?

Comment (kozlovic) [reply test/js_test.go:2755]: Sure. But if we were to add more options to Nak() (which is unlikely), then it may be weird too. Will change for now.

**Comment by kozlovic** [comment]
@derekcollison Updated in this commit: https://github.com/nats-io/nats.go/pull/894/commits/f1d50b84afd1c6cb3c7465f51343a66b6b0d407d

**Review by derekcollison — COMMENTED** [review]
In general LGTM, wondering if we can place nak delay into opts to avoid passing around explicit arg that is usually 0.

**Comment by kozlovic** [comment]
@derekcollison I think I was able to do as you asked: https://github.com/nats-io/nats.go/pull/894/commits/d34c12d8a5817d32d5c6ad043fe302bf8ddc1730

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**